### ISSUE TITLE
Travis CI build failing

### ISSUE THREAD

**Issue by rishipal** [google/closure-compiler#3669]
Last two travis builds are failing. 

[Build #5411](https://travis-ci.org/github/google/closure-compiler/builds/722137694)
[Build #5410](https://travis-ci.org/github/google/closure-compiler/builds/722075898) 

```
ERROR: /home/travis/build/google/[secure]/BUILD.bazel:202:1: Compiling 13 JavaScript files to compiler-j2cl_dev.js failed: Worker process returned an unparseable WorkResponse!
Did you try to print something to stdout? Workers aren't allowed to do this, as it breaks the protocol between Bazel and the worker process.
---8<---8<--- Exception details ---8<---8<---
com.google.protobuf.InvalidProtocolBufferException: While parsing a protocol message, the input ended unexpectedly in the middle of a field.  This could mean either that the input has been truncated or that an embedded message misreported its own length.
...
```

**Comment by lauraharker** [comment]
From https://github.com/google/closure-compiler/issues/3667, these warnings are possibly significant:

```
WARNING: Download from http://bazel-mirror.storage.googleapis.com/repo1.maven.org/maven2/com/google/errorprone/error_prone_annotations/2.3.2/error_prone_annotations-2.3.2.jar failed: class com.google.devtools.build.lib.bazel.repository.downloader.UnrecoverableHttpException GET returned 404 Not Found
WARNING: Download from http://bazel-mirror.storage.googleapis.com/repo1.maven.org/maven2/com/squareup/javapoet/1.13.0/javapoet-1.13.0.jar failed: class com.google.devtools.build.lib.bazel.repository.downloader.UnrecoverableHttpException GET returned 404 Not Found
```

**Comment by rishipal** [comment]
The following error in the log seems to indicate that `bazel build :all` is requesting high memory from the Travis (Trusty) VM. This could be because there have been multiple new targets added to the corresponding BUILD file. 
```
# Native memory allocation (mmap) failed to map 191889408 bytes for committing reserved memory.
```

I will try to reduce the overall memory usage by trying out the following:
1. Using `--discard_analysis_cache --nokeep_state_after_build --notrack_incremental_state` flags in travis.yml
2. Limiting the bazel build :all memory by `--host_jvm_args=-Xmx2g --host_jvm_args=-Xms2g`

Also, the travis.yml uses Trusty VM which is EOL'd - https://docs.travis-ci.com/user/reference/trusty/
Migrating to the newer Xenial VM might also resolve this.

**Comment by rishipal** [comment]
Started the PR #3672 to try out the first fix. [requesting a PR triggers the travis build]

**Comment by rishipal** [comment]
Started PR #3673 to try out the second fix.

**Comment by rishipal** [comment]
The build continues to fail with OOM with PR  #3672.

I've started PR #3675 to see if restricting Bazel jobs fixes the build. Since more targets are added to the BUILD file, it is possible that Bazel spawns large number of parallel jobs.  

Regarding migrating to the Xenial VM: Since we're considering switching to GitHub actions, migration from Trusty to Xenial might be futile.

**Comment by rishipal** [comment]
Building fewer targets fixes the Travis build with PR #3683.

**Comment by rishipal** [comment]
Closing as obsolete since we decided to move to using GitHub actions.
